### PR TITLE
Ava UI: `DownloadableContentManager` Refactor

### PR DESCRIPTION
**Pull request by IsaacMarovitz**
**Changes:**
- Switches to using compiled bindings
- Moves `DownloadableContentManager` to a content dialogue
- Updates layout to match rest of Ava UI

Screenshot_2023-01-15_at_19 37 11

### CODE REVIEW

File: Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+{
+    public class DownloadableContentManagerViewModel : BaseModel
+    {
+        private readonly List<DownloadableContentContainer> _downloadableContentContainerList;
+        private readonly string                             _downloadableContentJsonPath;
+
+        private VirtualFileSystem                           _virtualFileSystem;
+        private AvaloniaList<DownloadableContentModel>      _downloadableContents = new();
+        private AvaloniaList<DownloadableContentModel>      _views = new();
+        private AvaloniaList<DownloadableContentModel>      _selectedDownloadableContents = new();
```
Comment (AcK77) [inline Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs:37]:
```suggestion
        private VirtualFileSystem                      _virtualFileSystem;
        private AvaloniaList<DownloadableContentModel> _downloadableContents = new();
        private AvaloniaList<DownloadableContentModel> _views = new();
        private AvaloniaList<DownloadableContentModel> _selectedDownloadableContents = new();
```

File: Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+        private string _search;
+
```
Comment (AcK77) [inline Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs:40]:
```suggestion
```

File: Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+            _titleId   = titleId;
+            _titleName = titleName;
+
+            _downloadableContentJsonPath = Path.Combine(AppDataManager.GamesDirPath, titleId.ToString("x16"), "dlc.json");
+
+            try
+            {
+                _downloadableContentContainerList = JsonHelper.DeserializeFromFile<List<DownloadableContentContainer>>(_downloadableContentJsonPath);
+            }
+            catch
```
Comment (AcK77) [inline Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs:105]: A log message (or a dialog message) should be nice to have if the JSON file is corrupted

File: Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+
+                if (nca.Header.ContentType == NcaContentType.PublicData)
+                {
+                    if ((nca.Header.TitleId & 0xFFFFFFFFFFFFE000) != _titleId)
+                    {
+                        break;
+                    }
+
+                    var content = new DownloadableContentModel(nca.Header.TitleId.ToString("X16"), path,
+                        fileEntry.FullPath, true);
```
Comment (AcK77) [inline Ryujinx.Ava/UI/ViewModels/DownloadableContentManagerViewModel.cs:257]:
```suggestion
                    var content = new DownloadableContentModel(nca.Header.TitleId.ToString("X16"), path, fileEntry.FullPath, true);
```

**Review by AcK77 — CHANGES_REQUESTED** [review]
Lgtm except those following comments.

The search field is a bit strange (smaller than buttons so placeholder text is hidden) too:
image

**Review by marysaka — APPROVED** [review]

**Review by emmauss — APPROVED** [review]

**Review by TSRBerry — APPROVED** [review]
Thank you!

**Pull request merged**